src/parameters.py

```diff
@@ -1,124 +1,51 @@
+class TrainingParameters(BaseModel):
+    network: str = Field(description="type of dlsia network used")
+    num_classes: int = Field(description="number of classes as output channel")
+    num_epochs: int = Field(default=10, description="number of epochs")
+    optimizer: str = Field(default="Adam", description="optimizer used for training")
+    criterion: str = Field(default="CrossEntropyLoss", description="criterion for loss")
+    weights: Optional[str] = Field(default=None, description="weights per class for imbalanced labeling")
+    learning_rate: float = Field(default=1e-2, description='learning rate')
+    
+    activation: Optional[str] = Field(default="ReLU", description="activation function used in network")
+    normalization: Optional[str] = Field(default="BatchNorm2d", description="normalization used between layers")
+    convolution: Optional[str] = Field(default="Conv2d", description="convolution used in network")
 
-class DLSIANetwork(str, Enum):
-    msdnet = 'MSDNet'
-    tunet = 'TUNet'
-    tunet3plus = 'TUNet3+'
-
-
-class Optimizer(str, Enum):
-    adadelta = "Adadelta"
-    adagrad = "Adagrad"
-    adam = "Adam"
-    adamw = "AdamW"
-    sparseadam = "SparseAdam"
-    adamax = "Adamax"
-    asgd = "ASGD"
-    lbfgs = "LBFGS"
-    rmsprop = "RMSprop"
-    rprop = "Rprop"
-    sgd = "SGD"
-
-
-class Criterion(str, Enum):
-    l1loss = "L1Loss" 
-    mseloss = "MSELoss" 
-    crossentropyloss = "CrossEntropyLoss"
-    ctcloss = "CTCLoss"
-    poissonnllloss = "PoissonNLLLoss"
-    gaussiannllloss = "GaussianNLLLoss"
-    kldivloss = "KLDivLoss"
-    bceloss = "BCELoss"
-    bcewithlogitsloss = "BCEWithLogitsLoss"
-    marginrankingloss = "MarginRankingLoss"
-    hingeembeddingloss = "HingeEnbeddingLoss"
-    multilabelmarginloss = "MultiLabelMarginLoss"
-    huberloss = "HuberLoss"
-    smoothl1loss = "SmoothL1Loss"
-    softmarginloss = "SoftMarginLoss"
-    multilabelsoftmarginloss = "MutiLabelSoftMarginLoss"
-    cosineembeddingloss = "CosineEmbeddingLoss"
-    multimarginloss = "MultiMarginLoss"
-    tripletmarginloss = "TripletMarginLoss"
-    tripletmarginwithdistanceloss = "TripletMarginWithDistanceLoss"
-
-
-class Activation(str, Enum):
-    relu = "ReLU"
-    sigmoid = "Sigmoid"
-    tanh = "Tanh"
-    softmax = "Softmax"
-
+    # Below are Data Loader related parameters
+    shuffle_train: Optional[bool] = Field(default=True, description="whether to shuffle data in train set")
+    batch_size_train: Optional[int] = Field(default=1, description="batch size of train set")
 
-class Normalization(str, Enum):
-    batchnorm2d = "BatchNorm2d"
-    batchnorm3d = "BatchNorm3d"
+    shuffle_val: Optional[bool] = Field(default=True, description="whether to shuffle data in validation set")
+    batch_size_val: Optional[int] = Field(default=1, description="batch size of validation set")
 
+    shuffle_inference: Optional[bool] = Field(default=False, description="whether to shuffle data during inference")
```

Comment:
I agree on the inference side, as we want the result been produced in order. However, I believe having shuffle in validation will still be giving some merit in terms of reducing the model to learn from certain patterns (from a validation perspective)?

Happy to discuss more about this before I make modifications.

Reply:
I think shuffling the validation set could be problematic as well. Since changes in the validation accuracy could change because of shuffling instead of overall model performance (since validation loss is calculated per batch and then averaged).
This can be updated in the next pull request if preferred.